### ISSUE TITLE
IFFMpegProgressData compile error

### ISSUE THREAD

**Issue by hados99** [legraphista/ffmpeg-progress-wrapper#6]
Hi, I really appreciate supporting typescript.
but I have compile time error as below

```
> tsc --pretty

node_modules/ffmpeg-progress-wrapper/dist/index.d.ts:25:5 - error TS2411: Property 'speed' of type 'number | undefined' is not assignable to string index type 'string | number'.

25     speed?: number;
       ~~~~~


node_modules/ffmpeg-progress-wrapper/dist/index.d.ts:26:5 - error TS2411: Property 'eta' of type 'number | undefined' is not assignable to string index type 'string | number'.

26     eta?: number;
```

I googled ts2411 error, and saw some stackoverflow question.
https://stackoverflow.com/questions/33818178/typescript-interface-array-type-error-ts2411

Do you have any idea?
Is it necessary ```[s: string]: string | number;``` indexer?

**Comment by legraphista** [comment]
I believe you are running your TS configuration in strict.
Since all those properties are optional, it makes sense for TS to warn about them possibility being undefined too.
Unfortunately removing the indexer will break backwards compatibility.

I've added `undefined` to the indexer of IFFMpegProgressData
https://github.com/legraphista/ffmpeg-progress-wrapper/blob/e94fc588e165dfa426a8ff50717b9cd1bb3481cd/src/index.ts#L40

Can you please test master? 
You just need to replace the version in your package.json with `git+https://github.com/legraphista/ffmpeg-progress-wrapper.git`

**Comment by hados99** [comment]
I have not thought of strict config.
I tested master branch and it works well.
really thank you!!

**Comment by legraphista** [comment]
Good to hear that it's working. 
I will push a new npm version in a few hours.

**Comment by legraphista** [comment]
Published with version `1.1.3`
